org/mozilla/jss/pkcs11/PK11Signature.java

```diff
@@ -317,12 +334,47 @@ native protected boolean engineVerifyNative(byte[] sigBytes)
     public void engineSetParameter(AlgorithmParameterSpec params)
         throws InvalidAlgorithmParameterException, TokenException
     {
-        Assert.notYetImplemented("PK11Signature.engineSetParameter");
+        //For now we only care about RSAPSS
+        if( isRSAPSSAlgorithm((SignatureAlgorithm) algorithm) ) {
+            digestAlgorithm = getRSAPSSDigestAlgFromSpec((PSSParameterSpec) params);
+        }
+    }
+
+    private Algorithm getRSAPSSDigestAlgFromSpec(PSSParameterSpec params) {
+         String hashAlgName = params.getDigestAlgorithm();
+         Algorithm hashAlg = null;
+        if(hashAlgName.equals("SHA-256")) {
+            hashAlg =  DigestAlgorithm.SHA256;
+        }  else if(hashAlgName.equals("SHA-512")) {
+            hashAlg =  DigestAlgorithm.SHA512;
+        }  else if(hashAlgName.equals("SHA-384")) {
+            hashAlg =  DigestAlgorithm.SHA384;
+        } else {
+            hashAlg =  DigestAlgorithm.SHA256;
+        }
+
+        return hashAlg;
+    }
+
+    private boolean isRSAPSSAlgorithm(SignatureAlgorithm algorithm) {
+
+        if(algorithm == null) {
+            return false;
+        }
+        if( algorithm == SignatureAlgorithm.RSAPSSSignatureWithSHA256Digest
+            || algorithm == SignatureAlgorithm.RSAPSSSignatureWithSHA384Digest 
+            || algorithm == SignatureAlgorithm.RSAPSSSignatureWithSHA512Digest
+            || algorithm == SignatureAlgorithm.RSAPSSSignature) {
+
+            return true;
+        }
+        return false;
     }
 
 	protected PK11Token token;
 	protected TokenProxy tokenProxy;
 	protected Algorithm algorithm;
+        protected Algorithm digestAlgorithm;
```

Comment:
Could we keep the indentation the same? Thanks!

Reply:
There was a mixup of this and existing code. I just fixed them all in that section.